vignettes/download_mail.Rmd

```diff
@@ -0,0 +1,215 @@
+- **project-users**: For general user questions and discussions.
+
+Mod Mbox archives also organize mailing lists by topic. The apache mailing list archives can be found at https://lists.apache.org/.
+
+# Project Configuration File 
+
+Mailing List archives are hosted by their respective open source projects. Therefore, in order to use Kaiaulu downloaders to obtain mail data, you will need to access the respective open source project, and find out the URL tied to the archive you are interested in. Generally, that is the developer mailing list, if your interest is to understand communication patterns among developers. Alternatively, if the focus of the research is Q&A from the user base, then a user mailing list may make more sense.
+
+Because project lifetime can go as far as a few decades, to have the full picture of what communication took place in the project you may need to download multiple archives and combine them, after turning them into tables using the Kaiaulu parser.
+
+The information you need to find out for each open source project is documented in Kaiaulu using a project configuration file format. For pipermail and mod_mbox this is as follows:
+
+```
+mailing_list:
+  # for pipermail
+  pipermail:
+    project_key_1:
+      mailing_list: https://mta.openssl.org/pipermail/openssl-users/
+      save_folder_path: ../../rawdata/helix/pipermail/save_mbox_mail
+  # for mod mbox
+  mod_mbox:
+    project_key_1:
+      mailing_list: https://lists.apache.org/list.html?announce@apache.org
+      save_folder_path: ../../rawdata/helix/mod_mbox/save_mbox_mail
+```
+
+The most manual time intensive step you will be required is to locate the URL of the mailing list archive you wish for in the project website. This is specified under `mailing_list`. Note for pipermail this URL should point to the page containing links to the monthly archives (e.g. https://mta.openssl.org/pipermail/openssl-users/), not the top-level mailing list page that contains all the different types of archives (e.g. https://mta.openssl.org/mailman/listinfo/).
+
+
+Regardless of which mail archive you choose, the downloaders will store the mail data in monthly files, in a `.mbox` format. This is a simple text file that contains some markings to identify the header of the e-mail containing title, authors, etc. While you can open any of the .mbox downloaded files with any text editor, Kaiaulu parsers will format them into tables, as demonstrated below. 
+
+## Tools Configuration
+
+In addition to the mailing list configurations, you need to specify the path to the [Perceval](https://github.com/chaoss/grimoirelab-perceval) binary in tools.yml. See the wiki for further details on how to setup third party tools. 
+
+```{r}
+# Load tools configuration
+tools <- parse_config("../tools.yml")
+parse_perceval_path <- get_tool_project("perceval", tools)
+
+# Load project configuration
+conf <- parse_config("../conf/helix.yml")
+mbox_file_path <- get_mbox_path(conf, "project_key_1")
+```
```

Comment:
Currently, folders need to be manually created. I believe this will be resolved later, but right now this remains an inconvenience for the user. The function to create these folders may be found in the PR #333.